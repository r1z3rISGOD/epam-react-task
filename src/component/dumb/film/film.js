import React from 'react'
import { SearchHeadingNetflix } from '../search-heading-netflix/search-heading-netflix'
import './film.scss'
import { useHistory } from 'react-router-dom'
import { clearStore } from '../../../store/actions'
import { useDispatch } from 'react-redux'
import PropTypes from 'prop-types'

export const Film = (props) => {
  console.log(props)
  const dispatch = useDispatch()
  const history = useHistory()
  const { openedFilm } = props
  const date = openedFilm.release_date
  return (
        <div className='film'>
            <SearchHeadingNetflix>
                <button onClick={() => {
                  history.push('/')
                  dispatch(clearStore())
                  console.clear()
                }} className='film__search-button'>SEARCH</button>
            </SearchHeadingNetflix>
            <div className='film-body'>
                <img className='film-body__img' src={openedFilm.poster_path} alt={openedFilm.title}/>
                <div className='film-data'>
                    <div className='film-data__title-rating'>
                        <span className='film-data__title'>{openedFilm.title}</span>
                        <span className='film-data__rating'>{openedFilm.vote_average}</span>
                    </div>
                    <div className='film-data__date-runtime'>
                        <span className='film-data__date'>{date.substr(0, 4)}</span>
                        <span className='film-data__time'>{`${openedFilm.runtime} min`}</span>
                    </div>
                    <span className='film-data__overview'>{openedFilm.overview}</span>
                </div>
            </div>
        </div>
  )
}

Film.propTypes = {
  openedFilm: PropTypes.object
}
