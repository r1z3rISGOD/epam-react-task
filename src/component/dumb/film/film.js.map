{"version":3,"file":"film.js","sourceRoot":"","sources":["film.tsx"],"names":[],"mappings":";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;AAAA,6BAA8B;AAC9B,+BAA2C;AAC3C,oEAAgE;AAChE,uBAAoB;AACpB,8CAA2C;AAC3C,0CAAuC;AACvC,2CAAyC;AACzC,4DAAkE;AAClE,6CAAwC;AACxC,kDAAiH;AACjH,oCAAkC;AAM3B,IAAM,IAAI,GAAyB,UAAC,EAAc;QAAZ,UAAU,gBAAA;IAC/C,IAAA,KAAkB,gBAAQ,CAAU,UAAU,KAAK,SAAS,CAAC,EAA5D,IAAI,QAAA,EAAE,OAAO,QAA+C,CAAA;IACnE,IAAM,QAAQ,GAAG,yBAAW,EAAE,CAAA;IAC9B,IAAM,QAAQ,GAAG,wBAAS,EAAE,CAAA;IAE5B,SAAe,QAAQ;;;;;4BACD,qBAAM,wBAAS,CAAC,QAAQ,CAAC,IAAI,CAAC,CAAC,EAAA;;wBAA7C,WAAW,GAAG,SAA+B;wBACnD,QAAQ,CAAC,8BAAoB,CAAC,WAAW,CAAC,CAAC,CAAA;wBAC3C,QAAQ,CAAC,oCAA0B,CAAC,WAAW,CAAC,MAAM,CAAC,CAAC,CAAC,CAAC,CAAC,CAAA;wBACnC,qBAAM,wBAAS,CAAC,WAAW,CAAC,MAAM,CAAC,CAAC,CAAC,EAAE,QAAQ,EAAE,cAAc,CAAC,EAAA;;wBAAlF,eAAe,GAAG,SAAgE;wBACxF,QAAQ,CAAC,gCAAsB,CAAC,eAAe,CAAC,IAAI,CAAC,CAAC,CAAA;wBACtD,OAAO,CAAC,KAAK,CAAC,CAAA;;;;;KACf;IAED,iBAAS,CAAC;QACR,IAAI,IAAI,EAAE;YACR,QAAQ,EAAE,CAAA;SACX;IACH,CAAC,CAAC,CAAA;IAEF,IAAM,OAAO,GAAG,IAAI;QAClB,CAAC,CAAC,oBAAC,eAAM,OAAE;QACX,CAAC,CAAC,6BAAK,SAAS,EAAC,MAAM;YACX,oBAAC,6CAAoB;gBACjB,oBAAC,wBAAU,OAAE,CACM;YACvB,oBAAC,oBAAQ,IACL,UAAU,EAAE,UAAU,GACxB,CACA,CAAA;IAChB,OAAO,CACD,iCACK,OAAO,CACN,CACX,CAAA;AACH,CAAC,CAAA;AAnCY,QAAA,IAAI,QAmChB"}