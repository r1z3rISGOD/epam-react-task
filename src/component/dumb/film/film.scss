@import "./src/_variables/variables.scss";

.film {
  width: 85%;
  margin: auto;
  padding-bottom: 50px;
}

.film-data__title-rating {
  display: flex;
  flex-direction: row;
  align-items: center;
  flex-wrap: wrap;
  margin-bottom: 30px;
}

.film-data__date-runtime {
  margin-bottom: 30px;
}

.film-body__img {
  width: 30%;
}

.film-body {
  display: flex;
  flex-direction: row;
}

.film-data {
  color: white;
  width: 50%;
  margin: 0 auto;
  text-align: justify;
}

.film-data__title {
  font-size: 3rem;
  font-family: $font;
  color: $color;
}

.film-data__rating {
  margin-left: 20px;
  font-size: 2rem;
  font-family: $font;
  border-radius: 50%;
  border: 1px solid greenyellow;
  color: greenyellow;
  padding: 10px;
  height: 40px;
  width: 40px;
  display: flex;
  justify-content: center;
  align-items: center;
}

.film-data__date {
  font-family: $font;
  font-size: 1.5rem;
  font-weight: bold;
}

.film-data__time {
  font-family: $font;
  font-size: 1.5rem;
  font-weight: bold;
  margin-left: 30px;
}

.film-data__overview {
  font-family: $font;
  font-size: 1.5rem;
}
