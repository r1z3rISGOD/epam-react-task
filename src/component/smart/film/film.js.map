{"version":3,"file":"film.js","sourceRoot":"","sources":["film.tsx"],"names":[],"mappings":";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;AAAA,6BAA8B;AAC9B,+BAA2C;AAC3C,4EAAwE;AACxE,uBAAoB;AACpB,sDAAmD;AACnD,kDAA+C;AAC/C,2CAAyC;AACzC,4DAAkE;AAClE,6CAAwC;AACxC,kDAK+B;AAC/B,4CAA0C;AAC1C,qDAA6C;AAMtC,IAAM,IAAI,GAAyB,UAAC,EAAc;QAAZ,UAAU,gBAAA;IAC/C,IAAA,KAAkB,gBAAQ,CAAU,UAAU,KAAK,SAAS,CAAC,EAA5D,IAAI,QAAA,EAAE,OAAO,QAA+C,CAAA;IACnE,IAAM,QAAQ,GAAG,yBAAW,EAAE,CAAA;IAC9B,IAAM,QAAQ,GAAG,wBAAS,EAAE,CAAA;IAC5B,IAAM,OAAO,GAAG,6BAAU,EAAE,CAAA;IAE5B,SAAe,QAAQ;;;;;4BACD,qBAAM,wBAAS,CAAC,QAAQ,CAAC,IAAI,CAAC,CAAC,EAAA;;wBAA7C,WAAW,GAAG,SAA+B;wBACnD,QAAQ,CAAC,8BAAoB,CAAC,WAAW,CAAC,CAAC,CAAA;wBAC3C,QAAQ,CAAC,oCAA0B,CAAC,WAAW,CAAC,MAAM,CAAC,CAAC,CAAC,CAAC,CAAC,CAAA;wBACnC,qBAAM,wBAAS,CAAC,WAAW,CAAC,MAAM,CAAC,CAAC,CAAC,EAAE,QAAQ,EAAE,cAAc,CAAC,EAAA;;wBAAlF,eAAe,GAAG,SAAgE;wBACxF,QAAQ,CAAC,gCAAsB,CAAC,eAAe,CAAC,IAAI,CAAC,CAAC,CAAA;wBACtD,OAAO,CAAC,KAAK,CAAC,CAAA;;;;;KACf;IAED,iBAAS,CAAC;QACR,IAAI,IAAI,EAAE;YACR,QAAQ,EAAE,CAAA;SACX;IACH,CAAC,CAAC,CAAA;IAEF,IAAM,OAAO,GAAG;QACd,OAAO,CAAC,IAAI,CAAC,GAAG,CAAC,CAAA;QACjB,QAAQ,CAAC,oBAAU,EAAE,CAAC,CAAA;QACtB,OAAO,CAAC,KAAK,EAAE,CAAA;IACjB,CAAC,CAAA;IAED,IAAM,OAAO,GAAG,IAAI;QAClB,CAAC,CAAC,oBAAC,eAAM,OAAE;QACX,CAAC,CAAC,6BAAK,SAAS,EAAC,MAAM;YACX,oBAAC,6CAAoB;gBACjB,oBAAC,wBAAU,IAAC,OAAO,EAAE,OAAO,GAAG,CACZ;YACvB,oBAAC,oBAAQ,IACL,UAAU,EAAE,UAAU,GACxB,CACA,CAAA;IAChB,OAAO,CACD,iCACK,OAAO,CACN,CACX,CAAA;AACH,CAAC,CAAA;AA1CY,QAAA,IAAI,QA0ChB"}