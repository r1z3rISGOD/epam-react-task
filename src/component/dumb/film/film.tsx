import * as React from 'react'
import { useEffect, useState } from 'react'
import { SearchHeadingNetflix } from '../search-heading-netflix'
import './film.scss'
import { FilmButton } from '../film-button'
import { FilmBody } from '../film-body'
import { useDispatch } from 'react-redux'
import { fetchData, fetchFilm } from '../../services/fetchService'
import { useParams } from 'react-router'
import { putFetchedArrayToStore, putOpenedFilmsGenreToStore, putOpenedFilmToStore } from '../../../store/actions'
import { Loader } from '../loader'

interface FilmProps {
    openedFilm: any[]
}

export const Film : React.FC<FilmProps> = ({ openedFilm }) => {
  const [load, setLoad] = useState<boolean>(openedFilm === undefined)
  const dispatch = useDispatch()
  const location = useParams()

  async function fetching () : Promise<void> {
    const fetchedFilm = await fetchFilm(location['id'])
    dispatch(putOpenedFilmToStore(fetchedFilm))
    dispatch(putOpenedFilmsGenreToStore(fetchedFilm.genres[0]))
    const newFetchedArray = await fetchData(fetchedFilm.genres[0], 'genres', 'release_date')
    dispatch(putFetchedArrayToStore(newFetchedArray.data))
    setLoad(false)
  }

  useEffect(() => {
    if (load) {
      fetching()
    }
  })

  const content = load
    ? <Loader/>
    : <div className='film'>
                <SearchHeadingNetflix>
                    <FilmButton/>
                </SearchHeadingNetflix>
                <FilmBody
                    openedFilm={openedFilm}
                />
            </div>
  return (
        <div>
            {content}
        </div>
  )
}
