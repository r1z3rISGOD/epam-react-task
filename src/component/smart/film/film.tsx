import * as React from 'react'
import { useEffect, useState } from 'react'
import { SearchHeadingNetflix } from '../../dumb/search-heading-netflix'
import './film.scss'
import { FilmButton } from '../../dumb/film-button'
import { FilmBody } from '../../dumb/film-body'
import { useDispatch } from 'react-redux'
import { fetchData, fetchFilm } from '../../services/fetchService'
import { useParams } from 'react-router'
import {
  clearStore,
  putFetchedArrayToStore,
  putOpenedFilmsGenreToStore,
  putOpenedFilmToStore
} from '../../../store/actions'
import { Loader } from '../../dumb/loader'
import { useHistory } from 'react-router-dom'

interface FilmProps {
    openedFilm: any[]
}

export const Film : React.FC<FilmProps> = ({ openedFilm }) => {
  const [load, setLoad] = useState<boolean>(openedFilm === undefined)
  const dispatch = useDispatch()
  const location = useParams()
  const history = useHistory()

  async function fetching () : Promise<void> {
    const fetchedFilm = await fetchFilm(location['id'])
    dispatch(putOpenedFilmToStore(fetchedFilm))
    dispatch(putOpenedFilmsGenreToStore(fetchedFilm.genres[0]))
    const newFetchedArray = await fetchData(fetchedFilm.genres[0], 'genres', 'release_date')
    dispatch(putFetchedArrayToStore(newFetchedArray.data))
    setLoad(false)
  }

  useEffect(() => {
    if (load) {
      fetching()
    }
  })

  const onClick = () => {
    history.push('/')
    dispatch(clearStore())
    console.clear()
  }

  const content = load
    ? <Loader/>
    : <div className='film'>
                <SearchHeadingNetflix>
                    <FilmButton onClick={onClick}/>
                </SearchHeadingNetflix>
                <FilmBody
                    openedFilm={openedFilm}
                />
            </div>
  return (
        <div>
            {content}
        </div>
  )
}
