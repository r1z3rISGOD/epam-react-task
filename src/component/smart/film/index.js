import { Film } from './film'
export { Film }
