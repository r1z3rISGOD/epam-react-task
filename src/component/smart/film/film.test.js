import { Film } from './film.tsx'
import React from 'react'
jest.mock('react-redux')

const film = {
  id: 412302,
  title: 'Gemini',
  tagline: '',
  vote_average: 10,
  vote_count: 3,
  release_date: '2018-03-30',
  poster_path: 'https://image.tmdb.org/t/p/w500/oIltQs7MPk7VQFG3DJfgC63mShU.jpg',
  overview: 'A heinous crime tests the complex relationship between a tenacious personal assistant and her Hollywood starlet boss. As the assistant travels across Los Angeles to unravel the mystery, she must stay one step ahead of a determined policeman and confront her own understanding of friendship, truth and celebrity.',
  budget: 0,
  revenue: 0,
  genres: [
    'Mystery',
    'Thriller'
  ],
  runtime: 92
}

jest.mock('react-router', () => ({
  ...jest.requireActual('react-router'),
  useParams: () => ({
    id: 11111
  }),
  useRouteMatch: () => ({ url: '/id' })
}))

const mockDispatch = jest.fn
jest.mock('react-redux', () => ({
  useDispatch: () => mockDispatch
}))
mockDispatch.mockImplementation(() => {})

describe('123321', () => {
  beforeEach(() => {
  })

  it('results related', () => {
    const comp = shallow(<Film openedFilm={film}/>)
    expect(comp).toMatchSnapshot()
  })
})
