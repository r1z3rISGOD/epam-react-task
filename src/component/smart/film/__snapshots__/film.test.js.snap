// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`123321 results related 1`] = `""`;
